minject/inject.py

```diff
@@ -91,6 +118,24 @@ def _is_type(key: "RegistryKey[T]") -> TypeGuard[Type[T]]:
     return isinstance(key, type)
 
 
+def _is_key_async(key: RegistryKey) -> bool:
+    """
+    Check whether a registry key is an async context manager
+    marked for initialization within the registry (@async_context).
+    """
+    if isinstance(key, str):
+        return False
+    elif isinstance(key, RegistryMetadata):
+        return key.is_async_context()
+    elif isinstance(key, type):
+        try:  # type: ignore
```

Comment:
Comments explaining the reasons for the `type: ignore` annotations would be helpful.

(Could we avoid the need for the `try`–`except` block by using `_get_meta()` or `_gen_meta()` here?)